hack/build-image/Dockerfile

```diff
@@ -30,7 +30,7 @@ RUN wget --quiet https://github.com/kubernetes-sigs/kubebuilder/releases/downloa
     chmod +x /usr/local/kubebuilder/bin/kubebuilder
 
 # get controller-tools
-RUN go install sigs.k8s.io/controller-tools/cmd/controller-gen@v0.12.0
+RUN go install sigs.k8s.io/controller-tools/cmd/controller-gen@v0.14.0
```

Comment:
why this upgrade of controller-gen?

Reply:
Please see the commit message:
```
It also updates controller-gen to v0.14.0 to fix an issue under new
versino of go.
More details see https://github.com/golang/go/issues/65637
```

Reply:
thank for the explanation!

would add that information to the PR description, it is more visible to reviewers